src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -51,20 +50,20 @@ export class ItemDisplayComponent implements OnInit, AfterViewInit, OnDestroy {
     const url = this.itemData?.item.url || '';
     // TODO get sToken
     const sToken = '';
-    this.setUrl(this.taskProxyManager.getUrl(url, sToken, 'http://algorea.pem.dev', 'task-'));
+    this.setUrl(getTaskUrl(url, sToken, 'http://algorea.pem.dev', 'task-'));
   }
 
   ngAfterViewInit(): void {
     if (this.iframe) {
       const iframe = this.iframe.nativeElement;
-      this.taskProxyManager.getTaskProxy(iframe, this.taskIframeLoaded.bind(this), false);
+      getTaskProxy(iframe).subscribe((task: Task) => this.taskIframeLoaded(task));
```

Comment:
You still have this cascade of callback `ngAfterViewInit` -> `taskIframeLoaded -> `taskLoaded` -> `setViews`. From what I understand, that's out of scope of this PR.

Reply:
Yes, that's out of scope of this PR. I didn't try to change it yet as anyway the PR focusing on item-display will change that code a lot and make something better.